src/Setup/BuildPlan.purs

```diff
@@ -20,21 +19,19 @@ import Effect.Exception (Error)
 import GitHub.Actions.Core as Core
 import Setup.Data.Key (Key)
 import Setup.Data.Key as Key
-import Setup.Data.Tool (Tool)
+import Setup.Data.Tool (Tool(..))
 import Setup.Data.Tool as Tool
+import Setup.Data.VersionField (VersionField(..))
 import Text.Parsing.Parser (parseErrorMessage)
 import Text.Parsing.Parser as ParseError
 
 -- | The list of tools that should be downloaded and cached by the action
-type BuildPlan = Array { tool :: Tool, version :: Version }
+type BuildPlan = Array { tool :: Tool, versionField :: VersionField }
 
 -- | Construct the list of tools that sholud be downloaded and cached by the action
 constructBuildPlan :: Json -> ExceptT Error Effect BuildPlan
 constructBuildPlan json = map Array.catMaybes $ traverse (resolve json) Tool.allTools
 
--- | The parsed value of an input field that specifies a version
-data VersionField = Latest | Exact Version
-
 -- | Attempt to read the value of an input specifying a tool version
```

Comment:
We should expand this documentation comment, because I had to re-read the implementation!

```suggestion
-- | Attempt to read the value of an input specifying a tool version. If the input is missing
-- | and is not required, then we return `Nothing`. If the input is present and the user has
-- | specified an exact version or the latest version then we return the parsed `VersionField`.
```